core/src/main/java/org/apache/iceberg/util/PropertyUtil.java

```diff
@@ -57,6 +67,19 @@ public static double propertyAsDouble(
     return defaultValue;
   }
 
+  public static int propertyTryAsInt(
+      Map<String, String> properties, String property, int defaultValue) {
+    String value = properties.get(property);
+    if (value != null) {
+      try {
+        return Integer.parseInt(value);
+      } catch (NumberFormatException e) {
+        return defaultValue;
```

Comment:
Nit: can we remove this line and fall to the same return later?

Reply:
I feel a bit weird about empty catch clause as hurt readability a bit, so I tried to invert the if condition so that we still have 3 return statement but reads better. what do you think?